pyam/core.py

```diff
@@ -955,3 +955,33 @@ def check_aggregate(df, variable, components=None, units=None,
                                   multiplier=multiplier)
         df.meta['exclude'] |= fdf.meta['exclude']  # update if any excluded
         return vdf
+
+
+def filter_by_meta(data, df, **kwargs):
+    """Add columns from IamDataFrame.meta table to a pd.DataFrame and apply
+    filters (optional)
+
+    Parameters
+    ----------
+    data: pd.DataFrame instance, index must include `['model', 'scenario']`
+        the DataFrame to which meta columns are to be joined
+    df: IamDataFrame instance
+        the IamDataFrame from which meta columns are joined
+    kwargs:
+        meta columns to be joined, where `col=...` applies filters
+        by the given arguments (using `utils.pattern_match()`) and `col=None`
+        joins the column without filtering
+    """
+    data = data.copy()
+    idx = data.index.copy()
+    data.index = data.index.droplevel(list(set(idx.names) - set(META_IDX)))
+    data = data.join(df.meta[list(kwargs.keys())])
+    data.index = idx
+
+    # filter by joined columns
+    keep = np.array([True] * len(data))
+    for col, values in kwargs.items():
+        if values is not None:
```

Comment:
I see here this enables `integer=None` below. But wouldn't it be easier to just not have `integer` in the call at all?

Reply:
I used the `column=None` as a shorthand for joining the meta column with applying a filter, similar to `slice(None)` of `pd.MultiIndex`.

Reply:
To explain the rationale again - it is often useful to have that information for further processing like `groupby()`.